src/source.ts

```diff
@@ -66,19 +77,20 @@ export abstract class Source {
 /**
  * Source of docker image deployment is a local image from a directory
  */
-class DirectorySource extends Source {
-  private path: string;
+class DockerImageAssetPropsSource extends Source {
+  private assetProps: ecr_assets.DockerImageAssetProps;
 
-  constructor(path: string) {
+  constructor(assetProps: ecr_assets.DockerImageAssetProps) {
     super();
-    this.path = path;
+    this.assetProps = assetProps;
   }
 
   public bind(scope: Construct, context: SourceContext): SourceConfig {
-
-    const asset = new ecr_assets.DockerImageAsset(scope, 'asset', {
-      directory: this.path,
-    });
+    const asset = new ecr_assets.DockerImageAsset(
+      scope,
+      'asset',
+      this.assetProps,
+    );
```

Comment:
```suggestion
    const asset = new ecr_assets.DockerImageAsset(scope, 'asset', this.assetProps);
```
nit: fits well on 1 line and easier to read